pkg/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,240 @@
+				// refused connection will not return error, as host is reachable,
+				// just actively refuses connections (e.g. port is blocked)
+				return nil
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+func (hc *HealthChecker) checkReachabilityItem(r netReachabilityItem) error {
+	target := r.Host + ":" + strconv.Itoa(r.Port)
+	conn, err := hc.toolkit.tcpDialer.Dial("tcp", target)
+	if err != nil {
+		return fmt.Errorf("error trying to connect to %s: %w", target, err)
+	}
+	if conn != nil {
+		if err = conn.Close(); err != nil {
+			return fmt.Errorf("error closing connection: %w", err)
+		}
+	}
+	return nil
+}
+
+// NetHealthcheckConfig is a struct that holds healtcheck config.
+type NetHealthcheckConfig struct {
+	Interfaces   []string              `yaml:"interfaces,omitempty"`
+	Reachability []netReachabilityItem `yaml:"reachability,omitempty"`
+	Timeout      string                `yaml:"timeout,omitempty"`
+}
+
+type netReachabilityItem struct {
+	Host string `yaml:"host"`
+	Port int    `yaml:"port"`
+}
+
+// LoadConfig loads healtcheck config from file.
+func LoadConfig(configFile string) (*NetHealthcheckConfig, error) {
+	config := &NetHealthcheckConfig{}
+
+	isMandatory := false
+	if val := os.Getenv(configEnv); val != "" {
+		isMandatory = true
+		configFile = val
+	}
+
+	configExists := false
+
+	read, err := os.ReadFile(filepath.Clean(configFile))
+	if err != nil {
+		if isMandatory {
+			return nil, fmt.Errorf("error opening config file: %w", err)
+		}
+	} else {
+		configExists = true
+	}
+
+	if configExists {
+		err = yaml.Unmarshal(read, &config)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshalling config file: %w", err)
+		}
+	}
+
+	return config, nil
+}
+
+// Toolkit is a helper structure that holds interfaces and functions used by HealthChecker.
+type Toolkit struct {
+	frr        frrInterface
+	linkByName func(name string) (netlink.Link, error)
+	tcpDialer  tcpDialerInterface
+}
+
+// NewHealthCheckToolkit returns new HealthCheckToolkit.
+func NewHealthCheckToolkit(frr frrInterface, linkByName func(name string) (netlink.Link, error), tcpDialer tcpDialerInterface) *Toolkit {
+	return &Toolkit{
+		frr:        frr,
+		linkByName: linkByName,
+		tcpDialer:  tcpDialer,
+	}
+}
+
+// NewTCPDialer returns new tcpDialerInterface.
+func NewTCPDialer(dialerTimeout string) tcpDialerInterface {
+	timeout, err := time.ParseDuration(dialerTimeout)
+	if err != nil {
+		seconds, err := strconv.Atoi(dialerTimeout)
+		if err != nil {
+			timeout = time.Second * defaultTCPTimeout
+		} else {
+			timeout = time.Second * time.Duration(seconds)
+		}
+	}
+	return &net.Dialer{Timeout: timeout}
+}
+
+// NewDefaultHealthcheckToolkit returns.
+func NewDefaultHealthcheckToolkit(frr frrInterface, tcpDialer tcpDialerInterface) *Toolkit {
+	return NewHealthCheckToolkit(frr, netlink.LinkByName, tcpDialer)
+}
+
+//go:generate mockgen -source=../../pkg/healthcheck/healthcheck.go -destination=../../pkg/healthcheck/mock/mock_healthcheck.go
```

Comment:
Can you try the following here?
```
//go:generate mockgen -destination ../mock/mock_healtheck.go . frrInterface,tcpDialerInterface
```
alternatively this could be in something like a `doc.go` file in the `mock` package so the command is closer to the generated code. In that case the `.` would need to be replaced by the full package path (`github.com/telekom/das-schiff-network-operator/pkg/healthcheck`)

Reply:
Done. :) Had to export the interfaces though.